Makefile

```diff
@@ -38,52 +37,67 @@ ALL = \
 	\
 	$(ISO_ROOT)/pillar/repositories.sls \
 	$(ISO_ROOT)/pillar/top.sls \
+	\
+	$(ISO_ROOT)/product.txt \
 
 
 default: all
 .PHONY: default
 
-all: iso
+all: $(ISO)
 .PHONY: all
 
 all-local: $(ALL) ## Build all artifacts in the build tree
 .PHONY: all-local
 
-$(ISO_ROOT)/bootstrap.sh: scripts/bootstrap.sh.in
+$(ISO_ROOT)/bootstrap.sh: scripts/bootstrap.sh.in $(ISO_ROOT)/product.txt
 	mkdir -p $(shell dirname $@)
 	rm -f $@
-	sed s/@VERSION@/$(VERSION)/g < $< > $@ || rm -f $@
+	sed s/@VERSION@/$(shell source $(ISO_ROOT)/product.txt && echo $$SHORT_VERSION)/g < $< > $@ || (rm -f $@; false)
 	chmod a+x $@
 
 $(ISO_ROOT)/salt/%: salt/%
 	mkdir -p $(shell dirname $@)
 	rm -f $@
 	cp -a $< $@
 
-$(ISO_ROOT)/pillar/top.sls: pillar/top.sls.in
+$(ISO_ROOT)/pillar/top.sls: pillar/top.sls.in $(ISO_ROOT)/product.txt
 	mkdir -p $(shell dirname $@)
 	rm -f $@
-	sed s/@VERSION@/$(VERSION)/g < $< > $@ || rm -f $@
+	sed s/@VERSION@/$(shell source $(ISO_ROOT)/product.txt && echo $$SHORT_VERSION)/g < $< > $@ || (rm -f $@; false)
 
 $(ISO_ROOT)/pillar/%: pillar/%
 	mkdir -p $(shell dirname $@)
 	rm -f $@
 	cp -a $< $@
 
+$(ISO_ROOT)/product.txt: scripts/product.sh VERSION FORCE
+	rm -f $@
+	mkdir -p $(shell dirname $@)
+	env \
+		VERSION_MAJOR=$(VERSION_MAJOR) \
+		VERSION_MINOR=$(VERSION_MINOR) \
+		VERSION_PATCH=$(VERSION_PATCH) \
+		VERSION_SUFFIX=$(VERSION_SUFFIX) \
+		$< > $@ || (rm -f $@; false)
+
+FORCE:
+
 clean: ## Clean the build tree
 	rm -rf $(BUILD_ROOT)
 .PHONY: clean
 
 iso: $(ISO) ## Build the MetalK8s ISO image
 .PHONY: iso
 
-$(ISO): all-local
+$(ISO): $(ALL)
+	source $(ISO_ROOT)/product.txt && \
```

Comment:
@slaperche-scality Did it this way, not sure what you meant with a 'target' to achieve this in a prior comment.

Reply:
Yup, that looks good to me.